usage/rest-server/src/main/java/brooklyn/rest/resources/ApplicationResource.java

```diff
@@ -82,6 +83,7 @@
 import com.google.common.collect.FluentIterable;
 import com.google.common.collect.Iterables;
 
+@HaHotStateRequired
```

Comment:
Both ways can be used, but having class-wide annotation is better in this case as it is too easy to miss the annotation in newly added methods. If we need to be more selective, then method-scope is implemented as well.